src/obj_dir.mli

```diff
@@ -1,7 +1,37 @@
 open! Stdune
 
+module Local : sig
+  type t
+
+  (** The source_root directory *)
+  val dir : t -> Path.Build.t
+
+  val make_exe: dir:Path.Build.t -> name:string -> t
+
+  val need_dedicated_public_dir : t -> bool
```

Comment:
Not related to this PR, but we'll need more comment in this module.  I can't remember what all these functions mean